src/epgsql_sock.erl

```diff
@@ -89,13 +91,20 @@
                 repl_feedback_required,
                 repl_cbmodule,
                 repl_cbstate,
-                repl_receiver}).
+                repl_receiver,
+                timer}).
 
 %% -- client interface --
 
 start_link() ->
     gen_server:start_link(?MODULE, [], []).
 
+command(C, Command) ->
+  gen_server:call(C, Command).
+
+timed_command(C, Command, Timeout) ->
+  gen_server:call(C, {timed, Command, Timeout}).
```

Comment:
It's definitely discussible, but we may imagine situation when `epgsql_sock`'s mailbox is overflowed and this `{timed ...` message stays in a mailbox for some significant amount of time. So, when it reaches `handle_call({timed...) -> erlang:start_timer(Timeout, ...)` we may have a lot of time already passed.

As a some kind of solution we may convert relative `Timeout` to absolute time before `gen_server:call` and then convert it back on arrival inside `handle_call`. But I'm not sure if it worth it.